### PR TITLE
 API for users to new model

### PR DESCRIPTION
**Pull request by reasonerjt**
This commit moves the legacy apis related to users to new model.
 Some funcs under common/dao are left b/c they are used by other module,
 which should also be shifted to leverage managers.
 We'll handle them separately.

 Signed-off-by: Daniel Jiang <jiangd@vmware.com>

### CODE REVIEW

File: src/pkg/user/dao/dao.go [hunk 1]
```diff
@@ -17,24 +17,75 @@ package dao
...
+		return 0, errors.BadRequestError(nil).WithMessage("user ID is set when creating user: %d", user.UserID)
+	}
+	ormer, err := orm.FromContext(ctx)
+	if err != nil {
+		return 0, err
+	}
+	id, err := ormer.Insert(user)
+	if e := orm.AsConflictError(err, "user %s or email %s already exists", user.Username, user.Email); e != nil {
+		err = e
+	}
```
Comment (heww) [inline src/pkg/user/dao/dao.go:70]: `WrapConflictError` will be better than `AsConflictError`

```go
if err != nil {
    return 0, orm.WrapConflictError(err, "user %s or email %s already exists", user.Username, user.Email)
}
```

File: src/server/v2.0/handler/user.go [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+func (u *usersAPI) SetCliSecret(ctx context.Context, params users.SetCliSecretParams) middleware.Responder {
+	uid := int(params.UserID)
+	if err := u.requireForCLISecret(ctx, uid); err != nil {
+		return u.SendError(ctx, err)
+	}
+	if err := requireValidSecret(params.Secret.Secret); err != nil {
+		return u.SendError(ctx, err)
+	}
+	if err := u.ctl.SetCliSecret(ctx, uid, params.Secret.Secret); err != nil {
+		logger.Errorf("Failed to set CLI secret, error: %v", err)
```
Comment (heww) [inline src/server/v2.0/handler/user.go:62]: `log.G(ctx).Errorf("Failed to set CLI secret, error: %v", err)`?

File: src/server/v2.0/handler/user.go [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+		return u.SendError(ctx, err)
+	}
+	return users.NewSetCliSecretOK()
+}
+
+func (u *usersAPI) CreateUser(ctx context.Context, params users.CreateUserParams) middleware.Responder {
+	logger := log.GetLogger(ctx)
```
Comment (heww) [inline src/server/v2.0/handler/user.go:69]: No need to get a logger here, it's used only once in this method at line 88. Just using `log.G(ctx).Errorf("Failed to create user, error: %v", err)` is ok.

File: src/server/v2.0/handler/user.go [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+	values := params.HTTPRequest.URL.Query()
+	for _, k := range []string{"username", "email"} {
+		if v := values.Get(k); v != "" {
+			query.Keywords[k] = &q.FuzzyMatchValue{Value: v}
+		}
+	}
+	total, err := u.ctl.Count(ctx, query)
+	if err != nil {
+		return u.SendError(ctx, err)
+	}
```
Comment (heww) [inline src/server/v2.0/handler/user.go:113]: May return directly here when the total is zero to reduce a DB query.

File: src/server/v2.0/handler/user.go [hunk 5]
```diff
@@ -0,0 +1,485 @@
...
+	sctx, _ := security.FromContext(ctx)
+	if ns, ok := types.NamespaceFromResource(rbac.Resource(scope)); ok {
+		for _, policy := range ns.GetPolicies() {
+			if sctx.Can(ctx, policy.Action, policy.Resource) {
+				policies = append(policies, policy)
+			}
+		}
+	}
+	var res []*models.Permission
+	relative := params.Relative != nil && *params.Relative
```
Comment (heww) [inline src/server/v2.0/handler/user.go:149]: `relative := lib.BoolValue(params. Relative)` ?

File: src/server/v2.0/handler/user.go [hunk 6]
```diff
@@ -0,0 +1,485 @@
...
+	}
+	return users.NewUpdateUserProfileOK()
+}
+
+func newUsersAPI() restapi.UsersAPI {
+	return &usersAPI{
+		ctl:     user.Ctl,
+		getAuth: config.AuthMode,
+	}
+}
```
Comment (heww) [inline src/server/v2.0/handler/user.go:261]: Move this func to the front or end?

File: src/server/v2.0/handler/user.go (hunk 2 above)
Comment (reasonerjt) [reply src/server/v2.0/handler/user.go:69]: fixed

File: src/server/v2.0/handler/user.go (hunk 3 above)
Comment (reasonerjt) [reply src/server/v2.0/handler/user.go:76]: fixed

File: src/server/v2.0/handler/user.go (hunk 4 above)
Comment (reasonerjt) [reply src/server/v2.0/handler/user.go:120]: good point, fixed.

File: src/server/v2.0/handler/user.go (hunk 5 above)
Comment (reasonerjt) [reply src/server/v2.0/handler/user.go]: Ah, thanks.  I'm not aware of this utility.

File: src/server/v2.0/handler/user.go (hunk 6 above)
Comment (reasonerjt) [reply src/server/v2.0/handler/user.go]: Sure.

**Review by Vad1mo — APPROVED** [review]

**Review by wy65701436 — APPROVED** [review]
lgtm

**Review by heww — APPROVED** [review]

**Pull request merged**